### PR TITLE
Add warning comment to dangling references docs.

### PR DESCRIPTION
**Pull request by ekpyron**
Should be fine to merge once one of the ``_cli`` jobs passes.

### CODE REVIEW

File: docs/types/reference-types.rst [hunk 1]
```diff
@@ -555,6 +555,7 @@ to long layout in storage<bytes-and-string>`.
     // SPDX-License-Identifier: GPL-3.0
     pragma solidity >=0.8.0 <0.9.0;
 
+    // This will report a warning
```
Comment (cameel) [inline docs/types/reference-types.rst:558]: I'd put it at the line that generates the warning (i.e. the one that does `push()`).

**Review by cameel — APPROVED** [review]
Approving anyway since it's not a big problem but I'd put the comment directly over the problematic line.

File: docs/types/reference-types.rst (hunk 1 above)
Comment (ekpyron) [reply docs/types/reference-types.rst:558]: All other instances have it up front, so I'll keep it there :-).

**Comment by ekpyron** [comment]
The relevant parts of the CI runs just passed, so I'll merge.

File: docs/types/reference-types.rst (hunk 1 above)
Comment (cameel) [reply docs/types/reference-types.rst:558]: Oh, ok then. I thought we were using it more like an integral part of the surrounding text. Apparently not :)

**Pull request merged**